smack-integration-test/src/main/java/org/igniterealtime/smack/inttest/Configuration.java

```diff
@@ -473,6 +480,15 @@ public Builder setCompatibilityMode(String compatibilityModeString) {
             return setCompatibilityMode(compatibilityMode);
         }
 
+        public Builder setTestRunResultProcessors(String testRunResultProcessorsString) {
+            if (testRunResultProcessorsString == null) {
```

Comment:
Calling a public API like this with `null` should usually throw an exception. It is not defined that calling this method with `null` actually does and therefore it is most likely a sign of a programming error, e.g., the user wanting to set a result processor for somehow it ended up being `null`.

The same argument can be made for the empty string.

Therefore,  use `StringUtils.requireNonNullNorEmpty(testRunResultProcessorString)` here.

Reply:
I'm happy to change this, but it's a pattern that's used throughout this implementation. Do you want me to change it everywhere?

Reply:
Uh, fair point. Thanks for pointing this out. There is probably more to it that I don't remember.

In any case, if this is going to get changed, it will be done outside of this PR for all methods.